app/router.js

```diff
@@ -767,7 +775,56 @@ define(function (require, exports, module) {
       }
     },
 
+    navigateToSubroute: function(type, callArguments, view) {
+      var parentRoute = _.last(this.routeHistory.stack);
+      var subroute = {
+        type: type,
+        callArguments: callArguments,
+        view: view
+      };
+
+      var subrouteId = parentRoute.subroutes.push(subroute) - 1;
+      this.routeHistory.subrouteId = subrouteId;
+
+      Backbone.history.history.pushState({
+        subrouteId: subrouteId
+      }, document.title, Backbone.history.root + Backbone.history.fragment);
+    },
+
+    checkUrlForSubroute: function(e) {
+      var newSubrouteId = (e.state) ? e.state.subrouteId : null;
+      var previousSubrouteId = this.routeHistory.subrouteId;
+      var subroutes = _.last(this.routeHistory.stack)['subroutes'];
+
+      if(!_.isNumber(previousSubrouteId) && !_.isNumber(newSubrouteId)) {
+        return;
+      }
+
+      if(_.isNumber(previousSubrouteId) && subroutes[previousSubrouteId]) {
+        var subroute = subroutes[previousSubrouteId];
+        if(subroute.type == 'modal') {
+          subroute.view.close();
+        }
+
+        if(subroutes[previousSubrouteId].type == 'overlay') {
+          this.removeOverlayPage(subroute.view);
+        }
+      }
+
+      this.routeHistory.subrouteId = newSubrouteId;
+    },
+
     initialize: function (options) {
+      window.onpopstate = function (event) {
```

Comment:
Could we change this to `window.addEventListener('popstate' ... )` to prevent any overriding issues in the future?